plugin.video.vrt.nu/resources/lib/kodiwrappers/kodiwrapper.py

```diff
@@ -91,6 +92,53 @@ def get_setting(self, setting_id):
     def open_settings(self):
         self._addon.openSettings()
 
+    def get_global_setting(self, setting):
+        json_result = xbmc.executeJSONRPC('{"jsonrpc": "2.0", "method": "Settings.GetSettingValue", "params": {"setting": "%s"}, "id": 1}' % setting)
+        return json.loads(json_result)['result']['value']
+
+    def get_proxies(self):
+        usehttpproxy = self.get_global_setting('network.usehttpproxy')
+        if usehttpproxy is False:
+            return dict()
+
+        httpproxytype = self.get_global_setting('network.httpproxytype')
+
+        if httpproxytype != 0:
+            title = self.get_localized_string(32061)
+            message = self.get_localized_string(32062)
+            self.show_ok_dialog(title, message)
```

Comment:
Please return None if it doesn't work. Don't let Kodi crash.

Reply:
Ok.

Reply:
The problem is that I don't know if **requests[socks]** is installed. So at least now it would work if it is installed. I opened a ticket to get this support added to the official **script.module.requests** package, I think that's the best option we have.

PS I am also adding this to other addons now, planning to add it to netflix as well. So hopefully the need for SOCKS support will increase due to this.

Reply:
I am going to merge as-is, we can add the SOCKS detection later.